lib/slack/real_time/stores/loaded.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Slack
+  module RealTime
+    module Stores
+      # Full store with initialization logic
+      class Loaded < Store
+        CONVERSATION_TYPES = {
+          channels: 'public_channel',
+          groups: 'private_channel',
+          ims: 'im',
+          mpims: 'mpim'
+        }.freeze
+
+        attr_reader :caches
+
+        def initialize(_attrs, options = {})
+          @caches =
+            if options[:caches].is_a?(Array) || options[:caches].nil?
+              unknown_caches = options[:caches].to_a.map(&:to_sym) - CACHES
+              raise ArgumentError, "Unknown caches: #{unknown_caches.join(', ')}" unless unknown_caches.empty?
+
+              Set.new(options[:caches]).freeze
+            elsif options[:caches] == :all
+              Set.new(CACHES).freeze
+            else
+              raise ArgumentError, 'Unexpected value for option `caches`. Expects :all or an array of caches'
+            end
+          super
+        end
+
+        # The client has successfully connected to the server.
+        # @see https://api.slack.com/events/hello
+        # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/hello.json
+        on :hello do |_data, client|
+          if caches.include?(:teams)
+            team_info = client.web_client.team_info
+            team = Models::Team.new(team_info.team)
+            teams[team.id] = teams.key?(team.id) ? teams[team.id].merge(team) : team
+          end
+
+          if caches.include?(:users) || caches.include?(:bots)
+            client.web_client.users_list do |users_list|
+              users_list.members.each do |user_data|
+                if user_data.is_bot? && caches.include?(:bots)
+                  bot = Models::Bot.new(user_data)
+                  bots[bot.id] = bots.key?(bot.id) ? bots[bot.id].merge(bot) : bot
+                elsif user_data.is_user? && caches.include?(:users)
+                  user = Models::User.new(user_data)
+                  users[user.id] = users.key?(user.id) ? users[user.id].merge(user) : user
+                end
+              end
+            end
+          end
+
+          if caches.include?(:channels) || caches.include?(:groups) || caches.include?(:ims) || caches.include?(:mpims)
+            types = CONVERSATION_TYPES.slice(*caches).values.join(',')
+            client.web_client.conversations_list(types: types) do |conversations|
+              conversations.channels.each do |channel_data|
+                if channel_data.is_mpim?
+                  mpim = Models::Mpim.new(channel_data)
+                  mpims[mpim.id] = mpims.key?(mpim.id) ? mpims[mpim.id].merge(mpim) : mpim
```

Comment:
I don't think there's much use for mpims in a bot, but I added it as an option for completeness and because it seems like mpims (that include the bot user) had previously been included in the list of channels returned in the `rtm.start` response (they're considered a type of channel)